### DISCUSSION TITLE
[Question] Icons

### DISCUSSION THREAD

**Discussion by AlexHedley** [statiqdev/Discussions#215] [Q&A]
Can you include icons for custom fonts like https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/

And then add them in markdown like

`:fontawesome-regular-face-laugh-wink:`

**Comment by daveaglick** [comment]
I think the correct answer here is _sort of_. Statiq draws a pretty defined line between what the engine cares about and what a layout/theme/front-end should be doing. Normally I'd say something like this is entirely a front-end concern and while Statiq could certainly generate sites that use whatever icon font someone wants to use, doing so would mean adding the correct HTML in a layout or page, which Statiq doesn't really care about other than to output it.

That said, the linked feature from Material for MkDocs is a little more involved and includes some coordination with the generator to add special Markdown processing. Statiq doesn't have this same Markdown support out of the box since it's non-standard (though Statiq does enable a small number of non-standard Markdown features that users have come to expect). [You can also add additional Markdig extensions](https://www.statiq.dev/guide/content-and-data/template-languages/markdown#adding-extensions) to the Statiq Markdown processor. Some of the available ones [are listed here](https://github.com/xoofx/markdig/blob/master/readme.md) and you can also write your own - [here's a great guide](https://khalidabuhakmeh.com/how-to-write-a-dotnet-markdig-extension-for-markdown-processing).

Of course, generating HTML at the site of the icon doesn't do much good without the necessary CSS includes, which Material for MkDocs appears to include by default for it's own Markdown extension. You'll need to include that yourself in your theme or layout.

In short, Statiq can certainly support something like this, but it's not out of the box (right now). I suspect this would be something a theme could integrate more generally (just like Material for MkDocs does as a theme for MkDocs), but at the moment you'd need to write a small Markdig extension to handle the special Markdown syntax and add some CSS and/or JavaScript to your layout to light it up.

**Reply by daveaglick** [reply]
BTW - I moved this from the CleanBlog repo to the discussions area since I think it's a good question and is probably more general than just that one theme. Adding something like this could be done in any theme (or even as a sort of "plug in" that could integrate with any theme).

**Reply by AlexHedley** [reply]
Thanks for the detailed response and good idea to move it here.
I’ll take a look at the links provided and see what is possible, cheers.

Markdown Extension - for example [EscapeAt](https://github.com/statiqdev/Statiq.Framework/tree/main/src/extensions/Statiq.Markdown/EscapeAt)?

**Comment by TheBrambleShark** [comment]
Just to share my $0.02. I ran into this recently and used a solution that I think worked rather nicely.

Statiq has a [shortcodes](https://www.statiq.dev/guide/content-and-data/shortcodes) feature which lets you build custom tags that work in both Razor and Markdown. I don't have the code in front of me, but I was able to write a FontAwesome shortcode which accepted a few named parameters - the icon name, the style (solid, regular, etc.), and the size - and rendered the final html using an `<i>` tag.

It is a different syntax than what you're looking for, but it may be a decent way to solve your problem.
